### PR TITLE
Generate AOT Profilers for Non Shell based apps

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

[maui-app_20220723_080303.speedscope.zip](https://github.com/dotnet/maui/files/9173812/maui-app_20220723_080303.speedscope.zip)

This is an interesting spot. We have [similar code](https://github.com/dotnet/maui/blob/8e7c711326f3943910149cede48fb3c6fdffa1e1/src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellSectionRenderer.cs#L112) in shell that doesn't generate these emits. 

image

BEFORE AOT
```bash
Displayed com.companyname.shellvsnonshell/crc6429b4b1e1f66c1224.MainActivity: +798ms
2022-07-22 16:32:50.111 1904-2203/? I/ActivityTaskManager: Displayed com.companyname.shellvsnonshell/crc6429b4b1e1f66c1224.MainActivity: +722ms
2022-07-22 16:32:53.787 1904-2203/? I/ActivityTaskManager: Displayed com.companyname.shellvsnonshell/crc6429b4b1e1f66c1224.MainActivity: +724ms
2022-07-22 16:32:57.543 1904-2203/? I/ActivityTaskManager: Displayed com.companyname.shellvsnonshell/crc6429b4b1e1f66c1224.MainActivity: +719ms
2022-07-22 16:33:01.191 1904-2203/? I/ActivityTaskManager: Displayed com.companyname.shellvsnonshell/crc6429b4b1e1f66c1224.MainActivity: +704ms
2022-07-22 16:33:04.759 1904-2203/? I/ActivityTaskManager: Displayed com.companyname.shellvsnonshell/crc6429b4b1e1f66c1224.MainActivity: +695ms
```

AFTER AOT
```bash
2022-07-22 16:36:15.849 1904-2203/? I/ActivityTaskManager: Displayed com.companyname.shellvsnonshell/crc6429b4b1e1f66c1224.MainActivity: +709ms
2022-07-22 16:36:20.888 1904-2203/? I/ActivityTaskManager: Displayed com.companyname.shellvsnonshell/crc6429b4b1e1f66c1224.MainActivity: +712ms
2022-07-22 16:36:24.970 1904-2203/? I/ActivityTaskManager: Displayed com.companyname.shellvsnonshell/crc6429b4b1e1f66c1224.MainActivity: +686ms
2022-07-22 16:36:29.264 1904-2203/? I/ActivityTaskManager: Displayed com.companyname.shellvsnonshell/crc6429b4b1e1f66c1224.MainActivity: +700ms
2022-07-22 16:36:33.183 1904-2203/? I/ActivityTaskManager: Displayed com.companyname.shellvsnonshell/crc6429b4b1e1f66c1224.MainActivity: +690ms
```

### Issues Fixed

Fixes #

### CODE REVIEW

File: .nuspec/maui.aotprofile.txt [hunk 1]
```diff
@@ -1671,33 +1645,30 @@ Methods:
...
-	Microsoft.Maui.Controls.ColumnDefinition Microsoft.Maui.Controls.DefinitionCollection`1<Microsoft.Maui.Controls.ColumnDefinition>:get_Item (int)
 	Microsoft.Maui.Controls.ColumnDefinitionCollection Microsoft.Maui.Controls.Grid:get_ColumnDefinitions ()
 	Microsoft.Maui.Controls.ControlTemplate Microsoft.Maui.Controls.TemplatedPage:get_ControlTemplate ()
-	Microsoft.Maui.Controls.DataTemplate Microsoft.Maui.Controls.BaseShellItem:CreateDefaultFlyoutItemCell (string,string)
 	Microsoft.Maui.Controls.DataTemplate Microsoft.Maui.Controls.Internals.DataTemplateExtensions:SelectDataTemplate (Microsoft.Maui.Controls.DataTemplate,object,Microsoft.Maui.Controls.BindableObject)
-	Microsoft.Maui.Controls.DataTemplate Microsoft.Maui.Controls.Platform.Compatibility.ShellFlyoutRecyclerAdapter:get_DefaultItemTemplate ()
-	Microsoft.Maui.Controls.DataTemplate Microsoft.Maui.Controls.Platform.Compatibility.ShellFlyoutRecyclerAdapter:GetDataTemplate (int)
-	Microsoft.Maui.Controls.DataTemplate Microsoft.Maui.Controls.Shell:Microsoft.Maui.Controls.IShellController.GetFlyoutItemDataTemplate (Microsoft.Maui.Controls.BindableObject)
```
Comment (jonathanpeppers) [inline .nuspec/maui.aotprofile.txt:1700]: I was looking at the methods that were removed here, did we maybe lose a layout pass that was being recorded before?

File: src/ProfiledAot/src/maui/MainPage.xaml.cs [hunk 2]
```diff
@@ -3,16 +3,39 @@ namespace maui;
...
 	{
 		CounterBtn.Text = await CommonMethods.Invoke();
+		_startCompleted= true;
+		LoadFlyoutPage();
+	}
+
+	void OnMainPageLoaded(object sender, EventArgs e)
+	{
+		_isLoaded = true;
+		LoadFlyoutPage();
```
Comment (jonathanpeppers) [inline src/ProfiledAot/src/maui/MainPage.xaml.cs:30]: Should we try a `BeginInvoke` or `Task.Delay(100)` here to see if we get more methods in the profile?

We can also increase this time if needed:

https://github.com/dotnet/maui/blob/83ae546b3add4c81580efdfbd550376eaaafcba0/src/ProfiledAot/src/Directory.Build.targets#L31

File: .nuspec/maui.aotprofile.txt (hunk 1 above)
Comment (PureWeen) [reply .nuspec/maui.aotprofile.txt]: Fixed
I also noticed that we lost some animation calls if you run on a simulator vs physical so I reran it on a physical device and seems mostly additive now.

**Review by jonathanpeppers — APPROVED** [review]
Looks OK now to me

**Pull request merged**